src/app/model/Queen.ts

```diff
@@ -0,0 +1,50 @@
+import { PiecePositions } from './PiecesPositions';
+import { BOARDSIZE } from './Constances';
+import { Piece } from './Piece';
+import { columns, Row, Side, Square, Vector } from './Types';
+import { PiecePossibleMoves } from './PiecePossibleMoves';
+
+export class Queen extends Piece implements PiecePossibleMoves {
+  constructor(side: Side) {
+    super(side);
+  }
+
+  possibleMoves(position: Square, board: PiecePositions): Square[] {
+    return [
+      ...this.lineMoves(board, position, { col: 0, row: -1 }),
+      ...this.lineMoves(board, position, { col: 0, row: 1 }),
+      ...this.lineMoves(board, position, { col: -1, row: 0 }),
+      ...this.lineMoves(board, position, { col: 1, row: 0 }),
+      ...this.lineMoves(board, position, { row: -1, col: -1 }),
+      ...this.lineMoves(board, position, { row: -1, col: 1 }),
+      ...this.lineMoves(board, position, { row: 1, col: 1 }),
+      ...this.lineMoves(board, position, { row: 1, col: -1 }),
+    ];
+  }
+
+  private lineMoves(board: PiecePositions, actualPosition: Square, vector: Vector): Square[] {
+    const nextSquare: Square = {
+      column: columns[columns.indexOf(actualPosition.column) + vector.col],
```

Comment:
Widze tutaj pewien problem, tzn. zastanawia mnie jak to działa. Co jesli `columns.indexOf(actualPosition.column) + vector.col` wyjdzie poza zakres :) ? Ze nie ma wartosci np. columns[8]. Co sie wtedy dzieje?

Reply:
![image](https://user-images.githubusercontent.com/41823689/106519719-025a7f00-64dc-11eb-8688-8d904521d6a5.png)
Tutaj :)

Reply:
![image](https://user-images.githubusercontent.com/41823689/106519746-0f776e00-64dc-11eb-99b3-6c4d53fca438.png)

Reply:
Bardzo sprytne rozwiązanie Tomka ;)